db/background_mgr.go

```diff
@@ -12,17 +12,26 @@ import (
 	"net/http"
 	"sync"
 	"sync/atomic"
+	"time"
 
+	sgbucket "github.com/couchbase/sg-bucket"
 	"github.com/couchbase/sync_gateway/base"
 	"github.com/google/uuid"
 )
 
 type BackgroundProcessState string
 
+// These states are used for background tasks
+// Running = The process is currently doing work
+// Stopped = The process stopped following completion of its task
+// Aborting = A user has requested that the process be stopped and therefore will stop shortly (usually after completing its current 'iteration')
+// Aborted = The process has stopped either by user request or the process 'crashed' midway through --> Essentially means the process is not running and the previous run had not completed
```

Comment:
If it's necessary to differentiate between a requested stop and completion, I'd prefer to leave stopped and stopping as-is, and have a 'completed' state for processes that stopped following completion of its task.

Reply:
Wouldn't having a completed state be confusing if for example attachment compaction has a 'completed' state but resync just said 'stopped'. Or do you mean have a separate 'completed' field?